app/assets/javascripts/discourse/app/components/post.gjs

```diff
@@ -0,0 +1,574 @@
+    );
+  }
+
+  get hasRepliesAbove() {
+    return this.repliesAbove?.isResolved && this.repliesAbove.value.length > 0;
+  }
+
+  get id() {
+    return `post_${this.args.post.post_number}`;
+  }
+
+  get isFromCurrentUser() {
+    return this.currentUser && this.currentUser.id === this.args.post.user_id;
+  }
+
+  get isReplyingDirectlyToPostAbove() {
+    return (
+      this.args.prevPost &&
+      this.args.post.reply_to_post_number === this.args.prevPost.post_number &&
+      this.args.post.id !==
+        this.args.post.topic?.postStream?.filterUpwardsPostID
+    );
+  }
+
+  get isReplyToTabDisplayed() {
+    return PostMetaDataReplyToTab.shouldRender(
+      {
+        post: this.args.post,
+        isReplyingDirectlyToPostAbove: this.isReplyingDirectlyToPostAbove,
+      },
+      null,
+      getOwner(this)
+    );
+  }
+
+  get repliesShown() {
+    return this.filteredRepliesView
+      ? this.filteredRepliesShown
+      : this.repliesBelow.length > 0;
+  }
+
+  get shouldShowTopicMap() {
+    if (this.args.post.post_number !== 1) {
+      return false;
+    }
+
+    const isPM = this.args.post.topic.archetype === "private_message";
+    const isRegular = this.args.post.topic.archetype === "regular";
+    const showWithoutReplies =
+      this.siteSettings.show_topic_map_in_topics_without_replies;
+
+    return applyValueTransformer(
+      "post-show-topic-map",
+      isPM ||
+        (isRegular &&
+          (this.args.post.topic.posts_count > 1 || showWithoutReplies)),
+      { post: this.args.post, isPM, isRegular, showWithoutReplies }
+    );
+  }
+
+  get staged() {
+    return (
+      this.args.post.id === -1 ||
+      this.args.post.isSaving ||
+      this.args.post.staged
+    );
+  }
+
+  @action
+  copyLink() {
+    // Copying the link to clipboard on mobile doesn't make sense.
+    if (this.site.mobileView) {
+      return this.share();
+    }
+
+    const post = this.args.post;
+    const postId = post.id;
+
+    let actionCallback = () => clipboardCopy(getAbsoluteURL(post.shareUrl));
+
+    // Can't use clipboard in JS tests.
+    if (isTesting()) {
+      actionCallback = () => {};
+    }
+
+    postActionFeedback({
+      postId,
+      actionClass: "post-action-menu__copy-link",
+      messageKey: "post.controls.link_copied",
+      actionCallback,
+      errorCallback: () => this.share(),
+    });
+  }
+
+  @action
+  async loadMoreReplies() {
+    const after = this.repliesBelow.length
+      ? this.repliesBelow.at(-1).post_number
+      : 1;
+
+    const replies = await this.store.find("post-reply", {
+      postId: this.args.post.id,
+      after,
+    });
+
+    replies.forEach((reply) => {
+      // the components expect a post model instance
+      const replyAsPost = this.store.createRecord("post", reply);
+      this.repliesBelow.push(replyAsPost);
+    });
+  }
+
+  @action
+  async expandFirstPost() {
+    this.expandedFirstPost = new TrackedAsyncData(
+      await this.args.post.expand()
+    );
```

Comment:
```suggestion
    this.expandedFirstPost = new TrackedAsyncData(
      this.args.post.expand()
    );
```
If you `await` the argument value, then you're not actually passing a promise into TrackedAsyncData. Instead, you're waiting for the promise to resolve, and then passing the result into `TrackedAsyncData`.